stockpile/roles/dmidecode/files/dmide_to_json.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+import re as _re
+from subprocess import check_output
+import json
+import sys
+
+def _parse(dmide_fullinput):
+    output_data = {}
+    #  Each record is separated by double newlines
+    dmide_fullinput = dmide_fullinput.replace('\n\t\t',' ')
+    split_output = dmide_fullinput.replace('\t','').split('\n\n')
+    if len(split_output)<=1:
+        sys.exit(1)
+    for record in split_output:
+        #split_lines = record.replace(':','\":\"').split('\n')
+        split_lines = record.split('\n')
+        for x in range(len(split_lines)):
```

Comment:
Is there a reason why you have to treat the array as one indexed by the total number of entries instead of operating on each line, moving it from one list to another as it is processed?

Reply:
It's the hierarchy, and moving between one list to another list was giving issues with the way i was doing string manip and building the dictionary.